code-challenges.test.js

```diff
@@ -14,56 +14,177 @@
+            // Create a let/local variable called eachWord where it uses the built-in method .map in order to map through the array. 
+            //.map will return a new array with the same length as the original
+            //What we want is the value of the array so we specially use value and its first index(0) to captilize the letter
+            //return the local variable
+
+const firstLetterCaps = (array) => {
+    let eachWord = array.map(value => {
+        return value[0].toUpperCase() + value.substring(1)
+    }) 
+    return eachWord
+}
 
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
+// const vowelTester1 = "learn"
 // Expected output: 1
-const vowelTester2 = "academy"
+// const vowelTester2 = "academy"
 // Expected output: 0
-const vowelTester3 = "challenges"
+// const vowelTester3 = "challenges"
 // Expected output: 2
 
+//Psuedo-Code for Jest Test:
+    // A describe method that lists the name of the function OR naming of the particular test.
+    // A test/it method, nested within the describe block, that in plain words, describe what the function does.
+    // An expect method, nested within the test block, calling on the indexFirstVowel() function, followed by the toEqual() matcher that checks the expected output of the function return.
+    // Make 3 expects for each of the const variables provided.
+
+describe("indexFirstVowel", () => {
+    const vowelTester1 = "learn"
+    const vowelTester2 = "academy"
+    const vowelTester3 = "challenges"
+    it("returns the index of the first vowel", () => {
+        expect(indexFirstVowel(vowelTester1)).toEqual(1)
+        expect(indexFirstVowel(vowelTester2)).toEqual(0)
+        expect(indexFirstVowel(vowelTester3)).toEqual(2)
+    })
+})
 
 // b) Create the function that makes the test pass.
+    // Pseudo-Code for Function:
+        //Input: string
+        //Output: the index of the first vowel that appears on the string. 
+        //Process: 
+            // Create a function called indexFirstVowel znd take in a string as a parameter. 
+            // Create a for loop where the starting condition starts at 0, and keeps running the loop until string.length, and use i++. 
+            // Inside the curly brackets of the function create an if statement where the conditional statement says if string[i] === all the vowels (aeiou). 
+            // Make a return statement where it take the indexOf the string[i]
+
+    const indexFirstVowel = (string) => {
+        for(let i = 0; i < string.length; i++) {
+            if(string[i] === "a" || string[i] === "e" || string[i] === "i" || string[i] === "o" || string[i] === "u") {
+                return string.indexOf(string[i])
+            }
+        } 
+    }     
+    
```

Comment:
Logic looks great!  Can you think of how to rewrite this with a higher-order function?